inst/shiny/modules/tab_visuals.R

```diff
@@ -0,0 +1,560 @@
+        textInput(ns("filename"), "Enter filename of choice:", ""),
+        actionButton(ns("go2"), "GO"),
+        footer = modalButton("Close")
+      ))
+    })
+
+    # alternatively: save locally
+    output$downloadsum_browser <- downloadHandler(
+      filename = function() {
+        paste("NCA_summary.csv", sep = "_")
+      },
+      content = function(file) {
+        write.csv(summary_stats(), file)
+      }
+    )
+
+    # TAB: Parameter Box plots ----------------------------------------------------
+
+    # Create formatted Box plot data: PKNCA + PP results, linking DOSEA + PPTESTCD
+    boxplotdata <- reactive({
+      group_columns <- unname(unlist(res_nca()$data$conc$columns$groups))
+
+      left_join(
+        res_nca()$result %>%
+          filter(
+            end == Inf | startsWith(PPTESTCD, "aucint")
+          ),
+        res_nca()$data$conc$data %>%
+          distinct(across(all_of(group_columns)), .keep_all = TRUE),
+        by = group_columns,
+        keep = FALSE
+      ) %>%
+        # Intervals should also be considered as differentiated options each
+        mutate(
+          PPTESTCD = ifelse(
+            startsWith(PPTESTCD, "aucint"),
+            paste0(PPTESTCD, "_", start, "-", end),
+            PPTESTCD
+          )
+        )
+    })
+
+    # select which parameter to box or violin plot
+    output$selectboxplot <- renderUI({
+      param_choices <- boxplotdata()$PPTESTCD %>% unique()
+
+      pickerInput(ns("selected_param_boxplot"), "Choose the parameter to display:",
+                  choices = param_choices,
+                  selected = param_choices[1],
+                  multiple = FALSE,
+                  options = list(`actions-box` = TRUE))
+
+    })
+
+    # Selector for the variables to segregate box plots in the x axis
+    output$select_xvars_boxplot <- renderUI({
+      pickerInput(
+        inputId = ns("selected_xvars_boxplot"),
+        label = "Select X grouping variables",
+        multiple = TRUE,
+        choices = intersect(names(boxplotdata()), names(data())),
+        selected = "DOSEA"
+      )
+    })
+
+    output$select_colorvars_boxplot <- renderUI({
+      pickerInput(
+        inputId = ns("selected_colorvars_boxplot"),
+        label = "Select coloring variables to differentiate boxplots",
+        multiple = TRUE,
+        choices = intersect(names(boxplotdata()), names(data())),
+        selected = "DOSNO"
+      )
+    })
+
+    observeEvent(list(input$selected_xvars_boxplot,
+                      input$selected_colorvars_boxplot), {
+                   xvar_options_list <- lapply(
+                     c(input$selected_xvars_boxplot, input$selected_colorvars_boxplot),
+                     \(id_var) paste(id_var, unique(boxplotdata()[[id_var]]), sep = ": ")
+                   )
+
+                   names(xvar_options_list) <- c(
+                     input$selected_xvars_boxplot,
+                     input$selected_colorvars_boxplot
+                   )
```

Comment:
suggestion: You can use `setNames` to make this a tiny bit nicer.
```suggestion
                   xvar_options_list <- lapply(
                     c(input$selected_xvars_boxplot, input$selected_colorvars_boxplot),
                     \(id_var) paste(id_var, unique(boxplotdata()[[id_var]]), sep = ": ")
                   ) |>
                     setNames(c(input$selected_xvars_boxplot, input$selected_colorvars_boxplot))
```